### PR TITLE
npm release actions

### PR DESCRIPTION
**Pull request by davidleger95**
This time it actually works. (Ignore my 24 commits pls 🙃)

### CODE REVIEW

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+      - name: Install dependencies
+        run: npm install
+      - name: Set registry
+        run: 'echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc'
+      - name: Publish pre-release of @manifoldco/mui-core 🤫
```
Comment (dangodev) [inline .github/workflows/prerelease.yml]: LMAO

File: .gitignore [hunk 2]
```diff
@@ -11,16 +8,27 @@ loader/
...
-.vscode/
 .sass-cache/
+.stencil/
 .versions/
+.vscode/
+coverage/
+dist/
+loader/
 node_modules/
+pkg/
```
Comment (dangodev) [inline .gitignore]: We don’t need `pkg` to be ignored any more

File: .gitignore [hunk 3]
```diff
@@ -11,16 +8,27 @@ loader/
...
+Thumbs.db
+UserInterfaceState.xcuserstate
+.env
 
-.stencil/
+# Manifold
+.manifold.yml
+
+# Stencil
+!www/favicon.ico
```
Comment (dangodev) [inline .gitignore]: I don’t know why this is excluded. This should come after `www` but again you don’t really need to commit this so I’d just say remove it

File: package.json [hunk 4]
```diff
@@ -15,15 +21,16 @@
   "collection:main": "dist/collection/index.js",
   "unpkg": "dist/mui-core/mui-core.js",
   "files": [
-    "dist/",
-    "loader/"
+    "dist/"
   ],
```
Comment (dangodev) [inline package.json]: `package.json.files` is the opposite of `.npmignore`. `files` starts with npm publishing nothing, except for what’s in here. `.npmignore` starts with npm publishing everything, except for the items excluded.

Yes it’s unclear how they interact, but IMO `.npmignore` is clearer in purpose. I’d just delete `"files"` if you can. Or if Stencil yells at you, keep it but put `"loader/"` back to make sure that it always publishes to npm too

**Comment by dangodev** [comment]
LGTM! But be wary of that `"files"` change (if it works and `loader/` is on npm then you’re good though)

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+      - name: Install dependencies
+        run: npm install
+      - name: Set registry
+        run: 'echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc'
+      - name: Publish new release of @manifoldco/mui-core 🚀
+        run: 'npm publish --tag latest'
```
Comment (sslotsky) [inline .github/workflows/release.yml]: How are we setting the version number? I assumed we'd be updating the `version` field in `package.json` with the tag, but I don't see that happening. Did I miss it?

Comment (dangodev) [reply .github/workflows/release.yml]: That happens in the `prepare` script, which happens on `npm install` and `npm publish`. Putting it there makes sure it happens every time, both in CI or if someone publishes manually

**Pull request merged**